### PR TITLE
nixos/os-release: add additional info

### PR DESCRIPTION
**Pull request by jopejoe1**
## Description of changes
Added a few more options to os-release from https://www.freedesktop.org/software/systemd/man/latest/os-release.html

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — APPROVED** [review]
Diff LGTM

File: nixos/modules/misc/version.nix [hunk 1]
```diff
@@ -22,21 +22,26 @@ let
...
       SUPPORT_URL = optionalString isNixos "https://nixos.org/community.html";
       BUG_REPORT_URL = optionalString isNixos "https://github.com/NixOS/nixpkgs/issues";
       ANSI_COLOR = optionalString isNixos "1;34";
       IMAGE_ID = optionalString (config.system.image.id != null) config.system.image.id;
       IMAGE_VERSION = optionalString (config.system.image.version != null) config.system.image.version;
-    } // lib.optionalAttrs (cfg.variant_id != null) {
-      VARIANT_ID = cfg.variant_id;
+      VARIANT = optionalString (cfg.variantName != null) cfg.variantName;
+      VARIANT_ID = optionalString (cfg.variant_id != null) cfg.variant_id;
+      DEFAULT_HOSTNAME = config.networking.hostName;
```
Comment (MinerSebas) [inline nixos/modules/misc/version.nix:44]:
```suggestion
      DEFAULT_HOSTNAME = config.networking.fqdnOrHostName;
```

Comment (jopejoe1) [reply nixos/modules/misc/version.nix]: sure

**Review by a-kenji — APPROVED** [review]

**Review by pinpox — APPROVED** [review]

**Comment by oneingan** [comment]
This PR provokes a cascade of errors if `networking.domain != null` but `networking.hostName == ""`. Hard to workaround in a system because the os-release content propagates into initrd stuff.

I made a possible fix here https://github.com/NixOS/nixpkgs/pull/357647

**Pull request merged**